### ISSUE TITLE
ClassProperty context not working if publicOnly set to true

### ISSUE THREAD

**Issue by admir86** [gajus/eslint-plugin-jsdoc#786]
## Expected behavior

Using `ClassProperty` context should report missing JSDoc on properties in exported classes, event if `publicOnly` is set.

## Actual behavior

If `publicOnly` is set to `true`, class properties without JSDoc are not reported.

## ESLint Config

```json
"jsdoc/require-jsdoc": [
      "error",
      {
        "contexts": ["ClassProperty"],
        "publicOnly": true
      }
    ]
```

## ESLint sample

```ts
export class InovaAutoCompleteComponent {
  public disabled = false;
}
```

## Full Config

```json
"jsdoc/require-jsdoc": [
      "error",
      {
        "contexts": [
          "MethodDefinition:not([accessibility=\"private\"]):not([override=true]) > FunctionExpression",
          "TSInterfaceDeclaration",
          "FunctionExpression",
          "FunctionDeclaration",
          "ClassProperty"
        ],
        "publicOnly": true,
        "require": {
          "ClassDeclaration": true,
          "ClassExpression": true,
          "ArrowFunctionExpression": true,
          "FunctionExpression": true
        },
        "checkConstructors": false,
        "checkGetters": "no-setter",
        "checkSetters": "no-getter"
      }
    ]
```

## Environment

- Node version: v14.17.6
- ESLint version: v7.22.0
- `eslint-plugin-jsdoc` version: 36.1.0

**Comment by brettz9** [comment]
Are you using `@typescript-eslint/parser`? I'm not able to replicate, as I am getting warnings with either of your settings using your code sample.

**Comment by brettz9** [comment]
We're seeing a problem with `ClassProperty` in `@typescript-eslint/parser` version 5.0.0+ but not with 4.x, so just a heads up that this may be related. Plan to investigate as have a chance.

**Comment by brettz9** [comment]
This might be related to #804. Try changing `ClassProperty` to `PropertyDefinition` (or adding both)

**Comment by admir86** [comment]
Unfortunately, this doesn't change anything

**Comment by brettz9** [comment]
What exact version of `@typescript-eslint/parser` do you have installed?

**Comment by admir86** [comment]
`4.33.0` this version is actually required from my current [nx](https://nx.dev/) version

**Comment by brettz9** [comment]
You might also try `Property`. If not, I'm a bit busy atm to investigate myself, but if that doesn't work for you, you could try using `@typescript-eslint/parser` programmatically against your code and logging the AST object that it builds to see what `type` of Node is present.

**Comment by admir86** [comment]
it doesn't work with `Property` either, at least not in classes. I don't know AST and `@typescript-eslint/parser`, it would take me too much time to get this up and running. Maybe it's fixed with a new version, I'm waiting for an update from nx.

**Comment by brettz9** [comment]
I think this sounds like a dupe of #841 . Although the example you gave reports a problem for us, when there is a comment on the class, it indeed does not report a problem. So with:

```js
/**
 *
 */
export class InovaAutoCompleteComponent {
  public disabled = false;
}
```

...in this example, with the comment above the class, the error will be suppressed for `public disabled = false` (which it shouldn't).

Is that possibly the problem you were seeing?

**Comment by admir86** [comment]
Unfortunately I can't confirm that.
If I remove the comment on the class, error is showing for missing class comment but not for the property:
![image](https://user-images.githubusercontent.com/36576424/157823400-2a282313-eb5d-4af1-aaea-1fa2a1f54511.png)

**Comment by brettz9** [comment]
Please update to the latest version and see if that fixes it for you.

**Comment by admir86** [comment]
just updated from `36.1.1` to `37.9.7` but still the same behaviour
this are my eslint dev deps:
![image](https://user-images.githubusercontent.com/36576424/157879890-f4b92101-7e28-49f7-9bfa-ec59cdea002d.png)
maybe it requires another package update?

**Comment by brettz9** [comment]
What happens if the comment is present on the class? You still see the error for the property?

**Comment by brettz9** [comment]
Have you recently tried `PropertyDefinition` within `contexts` to ensure the error triggers when the property comment is missing? With your example and the following reduced test case, I am able to get an error as expected when a comment is not present:

```js
{
      code: `
      /**
       *
       */
      export class InovaAutoCompleteComponent {
        public disabled = false;
      }
      `,
      options: [
        {
          contexts: [
            'PropertyDefinition',
          ],
          publicOnly: true,
        },
      ],
      parser: require.resolve('@typescript-eslint/parser'),
      parserOptions: {
        sourceType: 'module',
      },
    },
```

**Comment by brettz9** [comment]
If the above doesn't help you resolve, I expect we will need a minimal test repository to be able to reproduce.

**Comment by admir86** [comment]
> 



> Have you recently tried `PropertyDefinition` within `contexts` to ensure the error triggers when the property comment is missing? With your example and the following reduced test case, I am able to get an error as expected when a comment is not present:
> 
> ```js
> {
>       code: `
>       /**
>        *
>        */
>       export class InovaAutoCompleteComponent {
>         public disabled = false;
>       }
>       `,
>       options: [
>         {
>           contexts: [
>             'PropertyDefinition',
>           ],
>           publicOnly: true,
>         },
>       ],
>       parser: require.resolve('@typescript-eslint/parser'),
>       parserOptions: {
>         sourceType: 'module',
>       },
>     },
> ```

just tried out. Errors for properties without docs are now reported. But only if class docs is also missing. Fixing the missing class docs, fixes also the property missing docs error without having documented the property.

**Comment by admir86** [comment]
> If the above doesn't help you resolve, I expect we will need a minimal test repository to be able to reproduce.

I understand, but the problem is I don't know how to reproduce it 😄 I have tried to create a new nx repo with similar structure but in this newly created repo it worked as expected

**Comment by brettz9** [comment]
I am not familiar with the exact nature of your environment, but if you can't replicate on a new environment, it is possible you do want to remove your `node_modules` and reinstall.

> Errors for properties without docs are now reported. But only if class docs is also missing.

This should not be happening on the latest version, so I don't know if it is possible your environment somehow isn't actually installing the latest or has some caching issue.

It seems this is more of a support issue for `nx`, at least without a minimal test reproducible case, so closing. Maybe the likes of Stackoverflow could help. Hope it works out for ya!

**Comment by admir86** [comment]
damn I was in another branch (with the old plugin version) when tested with 'PropertyDefinition'. 😳 
It seems this fixed the problem, thanks very much  😊
